### PR TITLE
Fix critical damage not being applied before health & mana drain

### PR DESCRIPTION
**Pull request by Vulcanx**

### CODE REVIEW

File: src/combat.cpp [hunk 1]
```diff
@@ -492,8 +492,38 @@ void Combat::CombatHealthFunc(Creature* caster, Creature* target, const CombatPa
 		return;
 	}
 
+	Player* attackerPlayer = caster ? caster->getPlayer() : nullptr;
+	Player* targetPlayer = target ? target->getPlayer() : nullptr;
+
+	if (attackerPlayer && damage.primary.value < 0 || damage.secondary.value < 0) {
```
Comment (nekiro) [inline src/combat.cpp]:
```
if (attackerPlayer && (damage.primary.value < 0 || damage.secondary.value < 0)) {
```

File: src/combat.cpp [hunk 2]
```diff
@@ -510,8 +540,40 @@ void Combat::CombatManaFunc(Creature* caster, Creature* target, const CombatPara
 {
 	assert(damage);
 	CombatDamage damageCopy = *damage;
+
+	Player* attackerPlayer = caster ? caster->getPlayer() : nullptr;
+	Player* targetPlayer = target ? target->getPlayer() : nullptr;
+
+	if (attackerPlayer && damageCopy.primary.value < 0 || damageCopy.secondary.value < 0) {
```
Comment (nekiro) [inline src/combat.cpp]:
```
if (attackerPlayer && (damageCopy.primary.value < 0 || damageCopy.secondary.value < 0)) {
```

File: src/combat.cpp [hunk 3]
```diff
@@ -492,8 +492,38 @@ void Combat::CombatHealthFunc(Creature* caster, Creature* target, const CombatPa
...
+		if (chance != 0 && uniform_random(1, 100) <= chance) {
+			damage.primary.value += std::round(damage.primary.value * (skill / 100.));
+			damage.secondary.value += std::round(damage.secondary.value * (skill / 100.));
+			g_game.addMagicEffect(target->getPosition(), CONST_ME_CRITICAL_DAMAGE);
+		}
+	}
+
 	if ((damage.primary.value < 0 || damage.secondary.value < 0) && caster) {
-		Player* targetPlayer = target->getPlayer();
 		if (targetPlayer && caster->getPlayer() && targetPlayer->getSkull() != SKULL_BLACK) {
```
Comment (nekiro) [inline src/combat.cpp]:
```
if (targetPlayer && attackerPlayer && targetPlayer->getSkull() != SKULL_BLACK) {
```

**Comment by nekiro** [comment]
This is actually broken, criticals are now only applied to one target when you throw aoe combat spell.
You would need to move it to
```
Combat::doCombat(Creature* caster, const Position& position) const
```

**Comment by DSpeichert** [comment]
This is made obsolete by #2497, right?

**Comment by nekiro** [comment]
@DSpeichert 
Yes, this pull request is redundant.

**Pull request closed without merging**